Implementa dados antropométricos

msoares94 opened the pull request:
**DESCRIÇÃO:**

Implementação da funcionalidade especificada na [ISSUE-1035](https://github.com/portabilis/i-educar/issues/1035)

**Histórico de altura e peso**

<img width="1643" height="382" alt="image" src="https://github.com/user-attachments/assets/0f2ccf63-48fa-4995-9c2a-67f26d7daa93" />

**Histórico Antropométrico**
<img width="1655" height="578" alt="image" src="https://github.com/user-attachments/assets/af263454-128f-4756-9c4d-6351b8904ea9" />



**AMBIENTE:**

- Plataforma utilizada (Docker)
- Sistema operacional e versão (Fedora Linux 42)

msoares94 committed:
Implementação parcial da funcionalidade de dados antropométricos e demográficos

msoares94 committed:
Melhorando exibição dos dados antropométricos

msoares94 committed:
fix: ajustando exibição de valores

msoares94 committed:
fix: removendo código comentado

msoares94 committed:
Complementando implementação de Dados Antropométricos

msoares94 committed:
feat(aluno): implementa tabela histórica para dados antropométricos

msoares94 committed:
fix: 31.6% Duplication on New Code (required ≤ 10%)

msoares94 committed:
fix: unit test

msoares94 committed:
run composer format

msoares94 committed:
fix: WARN  Metadata found in doc-comment

msoares94 committed:
fix: 29.5% Duplication on New Code (required ≤ 10%)

msoares94 committed:
fix: 15.3% Duplication on New Code (required ≤ 10%)

msoares94 committed:
fix: Corrigindo erro dos seeders

msoares94 committed:
Reduzir duplicação adicional para menos de 10%

- Consolidar métodos dos seeders na classe abstrata
- Simplificar métodos de interpolação nos modelos usando trait genérico
- Consolidar métodos de geração HTML no educar_aluno_det.php
- Corrigir padrões de nome de arquivo nos seeders

msoares94 committed:
Finalizar redução de duplicação - consolidar métodos HTML

- Substituir gerarCelulasDadosBasicos e gerarCelulasClassificacao por gerarCelulasTabela único
- Eliminar duplicação na geração de HTML da tabela antropométrica
- Manter funcionalidade idêntica com menos código duplicado

msoares94 committed:
Corrigir issues identificados pelo SonarCloud

- Adicionar tipos de retorno em métodos (void, string, int, float)
- Substituir isset() com null coalescing operator onde apropriado
- Remover exit() e usar return em métodos void
- Adicionar tipo mixed para retorno de callback
- Melhorar type safety geral do código

msoares94 committed:
Resolver todos os 28 issues reportados pelo SonarCloud

- Definir constante DECIMAL_PRECISION para eliminar literal duplicado "decimal:6"
- Remover trailing whitespaces de todos os arquivos afetados
- Adicionar newlines no final dos arquivos
- Refatorar verifyDataFiles() para reduzir complexidade cognitiva de 18 para <15
- Remover return redundante em método void
- Reduzir número de returns em interpolateFromDataSet() para ≤3
- Criar XlsxException dedicada ao invés de Exception genérica
- Definir constantes TEST_BIRTH_DATE_* para eliminar literais duplicados em testes
- Melhorar type safety com tipos de retorno específicos

Todos os testes continuam passando (20 testes, 108 assertions)

msoares94 committed:
Resolver 25 issues finais reportados pelo SonarCloud

Issues de Formatação:
- Adicionar newlines no final de 6 arquivos seeder
- Remover trailing whitespace em AbstractAnthropometricSeeder
- Adicionar newline em XlsxException

Issues de Constantes:
- Definir NO_REFERENCE_DATA_LABEL e NOT_EVALUABLE_LABEL no AnthropometricService
- Adicionar constantes TEST_BIRTH_DATE_2020 e TEST_BIRTH_DATE_2024 nos testes
- Substituir todas as ocorrências por constantes (7 ocorrências)

Issues de Complexidade:
- Reduzir returns em obterClassificacaoIMCAdulto() de 6 para 2 usando array de classificações
- Consolidar processInterpolationBounds() de 4 para 3 returns
- Refatorar getInterpolatedData() e getInterpolatedField() para ≤3 returns cada
- Extrair processReferenceDate() para eliminar ternários aninhados
- Extrair performInterpolation() como método auxiliar
- Substituir normalizarSexo() por array map para reduzir returns

Issues de Switch Statement:
- Adicionar case default em AnthropometricDataSeeder

Resultado:
✅ Todos os 25 issues resolvidos
✅ Código mais limpo e manutenível
✅ 20 testes passando (108 assertions)
✅ Funcionalidade preservada

msoares94 committed:
Resolver as últimas 7 issues de métodos com múltiplos returns do SonarCloud

Refatoração final para atender aos requisitos de qualidade do código:
- calcularIMC(): extraído isValidIMCParameters()
- obterClassificacaoIMCCrianca(): extraído validateChildIMCParameters() e getChildIMCClassification()
- calcularEscoreZIMC(): extraído getLMSForZScore() e calculateZScore()
- obterRiscoCinturaAdulto(): simplificado com arrays de thresholds
- obterRiscoCinturaCrianca(): consolidado com operadores ternários
- interpolateFromDataSet(): extraído performDataSetInterpolation()
- getCompletenessStatus(): refatorado com variáveis booleanas

Todos os métodos agora possuem ≤3 returns conforme exigido.
Funcionalidade mantida - 20 testes passando com 108 asserções.

msoares94 committed:
Refatorar AnthropometricService para atender requisitos de qualidade do SonarCloud

Divisão da classe monolítica em componentes especializados:

• AnthropometricService (13 métodos)
  - Interface principal com métodos públicos
  - Orquestra as operações entre Calculator e DataLoader

• AnthropometricCalculator (15 métodos)
  - Cálculos matemáticos (IMC, Z-scores, interpolações)
  - Validações e normalizações de dados
  - Classificações antropométricas

• AnthropometricDataLoader (7 métodos)
  - Carregamento e cache de dados do banco
  - Interpolação específica de datasets LMS e P90
  - Isolamento da lógica de acesso a dados

Melhorias implementadas:
- Separação de responsabilidades seguindo Single Responsibility Principle
- Redução de 27 para 13 métodos na classe principal
- Remoção de trailing whitespaces
- Todos os métodos com ≤3 returns conforme SonarCloud
- Funcionalidade mantida: 20 testes passando com 108 asserções

msoares94 committed:
Corrigir as últimas 3 issues do SonarCloud

- Adicionar newlines no final dos arquivos AnthropometricCalculator.php e AnthropometricDataLoader.php
- Refatorar obterClassificacaoIMCCrianca() para reduzir de 4 para 2 returns
- Extrair método validateChildIMCInputs() para consolidar validações
- Remover trailing whitespaces

Funcionalidade mantida: 20 testes passando com 108 asserções.

edersoares commented on database/migrations/2025_08_22_000003_create_anthropometric_percentiles_table.php:
```diff
@@ -0,0 +1,54 @@
+     */
+    public function up(): void
+    {
+        Schema::create('pmieducar.anthropometric_percentiles', function (Blueprint $table) {
```
Pode ficar no schema `public` com o tempo queremos remover os demais schemas.

```suggestion
        Schema::create('anthropometric_percentiles', function (Blueprint $table) {
```

msoares94 replied:
Obrigado por informar, corrigido!

edersoares commented on database/migrations/2025_08_22_000003_create_anthropometric_percentiles_table.php:
```diff
@@ -0,0 +1,54 @@
+     */
+    public function down(): void
+    {
+        Schema::dropIfExists('pmieducar.anthropometric_percentiles');
```
```suggestion
        Schema::dropIfExists('anthropometric_percentiles');
```

edersoares commented on database/migrations/2025_08_22_000001_create_anthropometric_z_scores_table.php:
```diff
@@ -0,0 +1,48 @@
+     */
+    public function up(): void
+    {
+        Schema::create('pmieducar.anthropometric_z_scores', function (Blueprint $table) {
```
```suggestion
        Schema::create('anthropometric_z_scores', function (Blueprint $table) {
```

edersoares commented on database/migrations/2025_08_22_000001_create_anthropometric_z_scores_table.php:
```diff
@@ -0,0 +1,48 @@
+     */
+    public function down(): void
+    {
+        Schema::dropIfExists('pmieducar.anthropometric_z_scores');
```
```suggestion
        Schema::dropIfExists('anthropometric_z_scores');
```

edersoares commented on app/Services/AnthropometricDataLoader.php:
```diff
@@ -0,0 +1,142 @@
+            $hasWaistData = false;
+
+            // Carregar dados Z-scores do banco
+            if (class_exists('\\App\\Models\\AnthropometricZScore')) {
```
Essa validação faz sentido visto que a classe existirá no projeto?

msoares94 replied:
Obrigado por informar, corrigido!

edersoares commented on app/Services/AnthropometricDataLoader.php:
```diff
@@ -0,0 +1,142 @@
+            }
+
+            // Carregar dados de cintura P90 da tabela de percentis
+            if (class_exists('\\App\\Models\\AnthropometricPercentile')) {
```
Essa validação faz sentido visto que a classe existirá no projeto?

msoares94 replied:
Obrigado por informar, corrigido!

edersoares commented on app/Services/AnthropometricDataLoader.php:
```diff
@@ -0,0 +1,142 @@
+
+        } catch (\Exception $e) {
+            // Log do erro mas não falha - usa dados de fallback
+            if (function_exists('logger')) {
```
Função sempre irá existir

msoares94 committed:
fix: ajustar schema das tabelas antropométricas para public e remover validações desnecessárias

- Mover tabelas anthropometric_z_scores e anthropometric_percentiles do schema pmieducar para public
- Remover validações redundantes de class_exists() e function_exists() no AnthropometricDataLoader
- Simplificar código removendo verificações de classes/funções que sempre existem no contexto Laravel

msoares94 commented:
> Aproveitando eu fiz um teste e acredito que a implementação da classificação não está de acordo.

@edersoares, para entender melhor o seu teste, poderia me informar, data de nascimento e sexo do aluno(a);

_Os valores (1,80 m e 77–78 kg) parecem medidas de um adulto. O IMC aproximado (24) estaria certo para 1,80 m e 77,6 kg, mas pode ser incoerente com a idade de crianças._

msoares94 committed:
fix: ajustar schema das tabelas utilizadas nos models

msoares94 committed:
Para crianças < 5 anos, não há padrões estabelecidos de cintura

edersoares commented:
@Fred-A-P pra mim faz sentido essa funcionalidade, pode analisar por favor?
